frontend/src/components/helpers/auth/PersistentLogin.js

```diff
@@ -7,14 +7,31 @@ import { SocketMessages } from "../socket-messages/SocketMessages";
 import { GenericLoader } from "../../generic-loader/GenericLoader";
 import { PromptRun } from "../../custom-tools/prompt-card/PromptRun";
 
+let selectedProductStore;
+let selectedProduct;
+try {
+  selectedProductStore = require("../../../plugins/llm-whisperer/store/select-product-store.js");
+} catch {
+  // Ignore if hook not available
+}
+
 function PersistentLogin() {
   const [isLoading, setIsLoading] = useState(true);
   const { sessionDetails } = useSessionStore();
   const checkSessionValidity = useSessionValid();
 
+  try {
+    if (selectedProductStore?.useSelectedProductStore) {
+      selectedProduct = selectedProductStore?.useSelectedProductStore(
+        (state) => state?.selectedProduct
+      );
+    }
+  } catch (error) {
+    // Do nothing
+  }
```

Comment:
@jagadeeswaran-zipstack Here's the corresponding change to my previous suggestion,

```suggestion
  const selectedProduct = useSelectedProductStore
    ? useSelectedProductStore((state) => state?.selectedProduct)
    : null;
```